### DISCUSSION TITLE
d3d12va

### DISCUSSION THREAD

**Discussion by ghost** [hooke007/mpv_PlayKit#353] [mpv 通用]

**Comment by hooke007** [comment]
link一下对应fr https://github.com/mpv-player/mpv/issues/12071

开源社区对vulkan有谜一般的痴迷，官方愿意实现d3d12的概率和热情感觉很低

**Comment by tanyiok1234** [comment]
厉害了，借github仓库编译了一下，运行良好没遇到啥坑点。

**Comment by hingbong** [comment]
AMD平台，用`march=znver3`编译之后，运行良好

**Reply by hingbong** [reply]
![image](https://github.com/hooke007/MPV_lazy/assets/26398240/4e8be55a-6d15-4533-b90c-5de4cf8866a2)
之前看没有报错，以为可用，看了日志才知道不行

**Reply by hingbong** [reply]
![image](https://github.com/hooke007/MPV_lazy/assets/26398240/1e6a2afa-7de6-44e7-bd76-89651680db7f)
也是不行，d3d11就没问题

**Reply by hooke007** [reply]
用i或n卡，a卡估计要amd那边提交补丁

**Reply by tanyiok1234** [reply]
编译znver3倒不一定是zen3 处理器，[shinchiro](https://github.com/shinchiro)那个仓库升级gcc13以后移除了补丁[53e0d9a4df492d20e7f7d794934bc953116f0ef3]，但是gcc13实际没有修复这个问题，所以带了avx512就会闪退。
（PS: 要小改一下那个gcc12的补丁重新打回去才能用znver4编译, 不过好像没啥性能提升）
其实可以试下上面arm64-v9a佬编译的，我之前试过在amd cpu + N卡的情况下一样可以触发d3d12-copy。
或者下面这个是我因为好玩最近编译的march=znver4版本（不支持avx512可能会打不开？），A+N的配置下是没啥问题的。
<img width="694" alt="image" src="https://github.com/hooke007/MPV_lazy/assets/11721926/b686bd4a-3523-4f3d-9fc9-f9f94f4bf9c1">
https://drive.google.com/file/d/1NtSz1oWAezecTgWVK1cxk4df5vJd63ZB/view?usp=sharing

**Reply by tanyiok1234** [reply]
原来action里的那几个option真的是修这个问题的啊，没玩会，学习了。

**Reply by hingbong** [reply]
看表格，应该是不支持了
